cogs/admin.py

```diff
@@ -118,9 +118,9 @@ async def on_member_update(self, before, after):
                 await general_channel.send(f'{after.mention} has been promoted to {new_role}!\n'
                                            f'Promotes Flight Officers to Lieutenants as a group\n'
                                            f'Assassinate {scum.mention} to take the Commanders throne!')
-            else:
+            elif new_role.name == 'Retired':
                 await self.update_member_nick(None, after)
-        else:
```

Comment:
I believe len(after.roles) < len(before.roles) is required here. This checks that the users roles actually changed, otherwise it will call update_member_nick on all member updates, not just role updates.